### ISSUE TITLE
Missing semicolon doesn't warn for arrow functions in class properties

### ISSUE THREAD

**Issue by pablonete** [palantir/tslint#1542]
### Bug Report

- __TSLint version__: `3.15.1`
- __TypeScript version__: `1.8.10`
- __Running TSLint via__: VSCode

#### TypeScript code being linted

```ts
class A {
    private f = function() { return "f"; }
    private g = () => { return "g"; }
    private h = () => "h"
}
```

with `tslint.json` configuration:

```json
"semicolon": [true, "always"],
```

#### Actual behavior

Property f gets a warning *Missing semicolon*.
Properties g and h doesn't.

#### Expected behavior

Properties g and h should get the same warning.

I can understand that somebody may want to opt-out of this rule, but it seems inconsistent as is.

**Comment by JKillian** [comment]
This was an intentional change actually, to allow arrow functions to be declared like class methods: https://github.com/palantir/tslint/pull/1081 

In #1081, we decided that the complexity of adding another option wasn't worth it for this small edge case, sorry!
